tests/sys/psa_crypto_mac/test_hmac_sha384.c

```diff
@@ -73,207 +59,374 @@ static psa_status_t test_hmac_sha384(const unsigned char *key, size_t key_len,
     status = psa_import_key(&attributes, key, key_len, &key_handle);
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("Key import failed");
     }
 
     status = psa_mac_sign_setup(&operation, key_handle, PSA_ALG_HMAC(PSA_ALG_SHA_384));
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("MAC setup failed");
     }
 
-    status = psa_mac_update(&operation, message, message_len);
+    status = psa_mac_update(&operation, data, data_len);
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("MAC update failed");
     }
-    size_t hmac_length;
-    status = psa_mac_sign_finish(&operation, hmac, SHA384_DIGEST_LENGTH, &hmac_length);
+
+    status = psa_mac_sign_finish(&operation, hmac, sizeof(hmac), &hmac_length);
     if (status != PSA_SUCCESS) {
         psa_destroy_key(key_handle);
-        return status;
+        TEST_FAIL("MAC finishing failed");
     }
 
     psa_destroy_key(key_handle);
 
-    return PSA_SUCCESS;
+    TEST_ASSERT_MESSAGE(hmac_length == sizeof(hmac), "HMAC result size mismatch");
+    /* Using result_len is important to allow for truncating the checked data */
+    TEST_ASSERT_MESSAGE(memcmp(hmac, result, result_len) == 0, message);
```

Comment:
Shouldn't the first check really be `hmac_length == result_len`?

Reply:
This whould not allow for checking only a subset of the result, as done in case 5.

Reply:
I see, but then you should probably also make sure that `result_len <= hmac_length`.

Reply:
Ok